Lib/fontbakery/reporters/serialize.py

```diff
@@ -96,6 +96,11 @@ def _register(self, event):
             item['clustered']['value'] = value
       self._set_metadata(identity, item)
 
+    if check:
+      item['description'] = check.description
+      if item["key"][2] != ():
+        item['filename'] = self.runner.get_iterarg(*item["key"][2][0])
```

Comment:
It's pure speculation that filename will be here a at that index, made to fit to your task at hand. This makes the serialize reporter less generic.